### PR TITLE
Make generate_builtins path customizable for testing. Update builtins

### PR DESCRIPTION
**Pull request by IanButterworth**
Fixes https://github.com/JuliaDebug/JuliaInterpreter.jl/issues/549
Also updates builtins

@maleadt

### CODE REVIEW

File: test/check_builtins.jl [hunk 1]
```diff
@@ -5,8 +5,11 @@ using Test, DeepDiffs
     @testset "Check builtin.jl consistency" begin
         builtins_path = joinpath(@__DIR__, "..", "src", "builtins.jl")
         old_builtins = read(builtins_path, String)
-        include("../bin/generate_builtins.jl")
-        new_builtins = read(builtins_path, String)
+        new_builtins_path = tempname()
```
Comment (maleadt) [inline test/check_builtins.jl:8]: Maybe it's better to use `mktempdir()` and `$BUILTINS_DIR/builtins.jl` instead?

**Comment by maleadt** [comment]
LGTM, thanks!

**Comment by maleadt** [comment]
I'm not a member of JuliaDebug either. cc @KristofferC

**Comment by KristofferC** [comment]
This seems to fail everywhere :P. I think there might be some `@static isdefined` checks put into the builtin-generator?

**Comment by IanButterworth** [comment]
Spent too long there trying to engineer what `RECENTLY_ADDED` does. Should really read files from the top-down

File: bin/generate_builtins.jl [hunk 2]
```diff
@@ -316,4 +317,5 @@ end
 """)
 end
 
-generate_builtins(joinpath(@__DIR__, "..", "src", "builtins.jl"))
+builtins_dir = get(ENV, "BUILTINS_DIR", joinpath(@__DIR__, "..", "src"))
```
Comment (KristofferC) [inline bin/generate_builtins.jl:320]: Include `JULIAINTERPRETER`? Or `JULIA_JULIAINTERPRETER_BUILTINS_DIR` maybe.

**Pull request merged**